prow/github/hmac.go

```diff
@@ -46,3 +64,39 @@ func PayloadSignature(payload []byte, key []byte) string {
 	sum := mac.Sum(nil)
 	return "sha1=" + hex.EncodeToString(sum)
 }
+
+// extractHmacs returns all *valid* HMAC tokens for given repository/organization. It iterates over full tree as necessary (repo->org->global)
+func extractHmacs(repo string, tokenGenerator func() []byte) [][]byte {
+	t := tokenGenerator()
+	repoToTokenMap := map[string]HmacsForRepo{}
+
+	err := yaml.Unmarshal(t, &repoToTokenMap)
+	if err != nil {
+		logrus.WithError(err).Error("Couldn't unmarshal the hmac secret")
```

Comment:
This should probably actually return its error to be handled by its caller, instead of logging it.

(Its caller will probably then just log it, I guess)